onnxruntime/contrib_ops/rocm/bert/attention.cc

```diff
@@ -40,47 +46,44 @@ Status Attention<T>::ComputeInternal(OpKernelContext* context) const {
   const Tensor* mask_index = context->Input<Tensor>(3);
   const Tensor* past = context->Input<Tensor>(4);
   const Tensor* relative_position_bias = context->Input<Tensor>(5);
+  const Tensor* past_seq_len = context->Input<Tensor>(kPastSequenceLengthInputIndex);
 
   auto& device_prop = GetDeviceProp();
+  AttentionParameters parameters;
   ORT_RETURN_IF_ERROR(CheckInputs(input->Shape(),
                                   weights->Shape(),
                                   bias->Shape(),
                                   mask_index,
                                   past,
                                   relative_position_bias,
-                                  nullptr,
-                                  device_prop.maxThreadsPerBlock));
+                                  &parameters,
+                                  device_prop.maxThreadsPerBlock,
+                                  past_seq_len));
+  assert(parameters.sequence_length == parameters.kv_sequence_length);  // self attention 
 
-  // input shape (batch_size, sequence_length, input_hidden_size)
-  const auto& shape = input->Shape();
-  int batch_size = static_cast<int>(shape[0]);
-  int sequence_length = static_cast<int>(shape[1]);
-  int input_hidden_size = static_cast<int>(shape[2]);
-
-  // Note: Scenario where q_hidden_size == k_hidden_size != v_hidden_size is not supported in ROCM EP
-  // bias shape (3 * hidden_size)
-  const auto& bias_shape = bias->Shape();
-  int hidden_size = static_cast<int>(bias_shape[0]) / 3;
-
-  int head_size = hidden_size / num_heads_;
+  int batch_size = parameters.batch_size;
+  int sequence_length = parameters.sequence_length;
 
   TensorShapeVector output_shape(3);
-  output_shape[0] = shape[0];
-  output_shape[1] = shape[1];
-  output_shape[2] = static_cast<int64_t>(hidden_size);
+  output_shape[0] = static_cast<int64_t>(batch_size);
+  output_shape[1] = static_cast<int64_t>(sequence_length);
```

Comment:
Maybe omit the alias assigning and use `parameters.` exclusively?

Reply:
trying to be consistent with CUDA code, but I think it's fine.